lib/puppet/util/windows/api_types.rb

```diff
@@ -52,10 +53,10 @@ def read_handle
     alias_method :read_wchar, :read_uint16
     alias_method :read_word,  :read_uint16
 
-    def read_wide_string(char_length)
+    def read_wide_string(char_length, dst_encoding = Encoding.default_external)
       # char_length is number of wide chars (typically excluding NULLs), *not* bytes
       str = get_bytes(0, char_length * 2).force_encoding('UTF-16LE')
-      str.encode(Encoding.default_external)
+      str.encode(dst_encoding)
```

Comment:
I wonder if we should revisit 
 * `ADSI.self.computername` adsi.rb:44
 * `User.current_user_name` adsi.rb:240
 * `Error.format_error_code` error.rb:51
 * `File.resolve_symlink` file.rb:285

And make sure all of them come back correctly with weird default external encodings?